tests/test_check_parallel.py

```diff
@@ -0,0 +1,346 @@
+        actual = sorted(os.listdir('{}/{}'.format(self.test_dir, self.cache)))
+        expected = sorted(['000000.tar', 'index.db'])
+        self.assertEqualOrStop(actual, expected)
+
+    def testKeepTarsWithPreviouslySetHPSS_5(self):
+        """
+        Test `zstash check` in parallel when hpss is set in `zstash create`.
+
+        This time using self.test_dir using the default of 'zstash_test' 
+        (set in base.setUp()) AND changing directory to self.test_dir to 
+        create the zstash archive. Fails.
+
+        Bash commands:
+        # In zstash/tests directory
+        mkdir zstash_test
+        cd zstash_test
+        zstash create --hpss=none --maxsize 128 .
+        cd ..
+        ls zstash_test/zstash # Fails: does not produce expected output
+        cd zstash_test
+        zstash check --workers=2
+        cd ..
+        ls zstash_test/zstash
+        """
+        print_in_box('testKeepTarsWithPreviouslySetHPSS')
+        self.assertWorkspace()
+        if os.path.exists(self.test_dir):
+            shutil.rmtree(self.test_dir)
+        os.mkdir(self.test_dir)
+        zstash_path = ZSTASH_PATH
+        self.hpss_path = 'none'
+        # Run `zstash create`
+        os.chdir(self.test_dir)
+        run_cmd('{}zstash create --hpss={} --maxsize 128 .'.format(zstash_path, self.hpss_path))
+        os.chdir(TOP_LEVEL)
+        actual = sorted(os.listdir('{}/{}'.format(self.test_dir, self.cache)))
+        expected = sorted(['000000.tar', 'index.db'])
+        # Fails on this step with:
+        # AssertionError: ['index.db'] != ['000000.tar', 'index.db']
+        self.assertEqualOrStop(actual, expected)
+        # Run `zstash check` without specifying hpss
+        os.chdir(self.test_dir)
+        run_cmd('{}zstash check --workers=2'.format(zstash_path))
+        os.chdir(TOP_LEVEL)
+        actual = sorted(os.listdir('{}/{}'.format(self.test_dir, self.cache)))
+        expected = sorted(['000000.tar', 'index.db'])
+        self.assertEqualOrStop(actual, expected)
+
+
+    def testKeepTarsWithPreviouslySetHPSS_6(self):
+        """
+        Test `zstash check` in parallel when hpss is set in `zstash create`.
+
+        This time using self.test_dir using the default of 'zstash_test'
+        (set in base.setUp()) AND changing directory to self.test_dir to
+        create the zstash archive. Fails.
+
+        Bash commands:
+        # In zstash/tests directory
+        mkdir zstash_test
+        cd zstash_test
+        zstash create --hpss=none --maxsize 128 .
+        ls zstash # Fails: does not produce expected output
+        zstash check --workers=2
+        ls zstash
+        """
+        print_in_box('testKeepTarsWithPreviouslySetHPSS')
+        self.assertWorkspace()
+        if os.path.exists(self.test_dir):
+            shutil.rmtree(self.test_dir)
+        os.mkdir(self.test_dir)
+        zstash_path = ZSTASH_PATH
+        self.hpss_path = 'none'
+        # Run `zstash create`
+        #os.chdir(self.test_dir)
```

Comment:
Commenting this out makes this test pass. Why should it matter what directory we're in?